File: UberCalendarUI/Data/WebApiDataHandler.cs
using Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Text;
using System.Threading.Tasks;
using System.Web.Script.Serialization;

namespace UberCalendarUI.Data
{
    public class WebApiDataHandler : IDataHandler
    {
        List<CalendarEvent> events = new List<CalendarEvent>();
        WebClient wc = new WebClient();

        JavaScriptSerializer serializer = new JavaScriptSerializer();

        public WebApiDataHandler()
        {
            int BaseAdressPort = 8080;
            wc.BaseAddress = $"http://localhost:{BaseAdressPort}/UberCalendar/";
            int testPort = 50925;
            int IISPort = 8080;
        }


        public CalendarUser CredentialsCheck(CalendarUserCredentials credentials)
        {
            //
            //bug: this methods doesn't returns calendarUser
            //

            wc.Headers.Add("Content-Type", "application/json");
            wc.Headers.Add("Data-Type", "application/json");
            string userJson = serializer.Serialize(credentials);
            string test = wc.BaseAddress + "GetUser" + userJson;
            var loggedInUser =  wc.UploadString(wc.BaseAddress + "GetUser", "Post", userJson);
            // CalendarUser loggedInUser = serializer.Deserialize<CalendarUser>);
            //return loggedInUser;
            return new CalendarUser();
        }

        public List<CalendarEvent> GetEvents(DateTime dateForEvents, CalendarUser loggedInUser)
        {
            string message = wc.DownloadString($"{wc.BaseAddress}GetEventsForUser?date={dateForEvents}&userId={loggedInUser.UserID}");
            events.Clear();
            events = serializer.Deserialize<List<CalendarEvent>>(message);
            return events;
        }

        public void RegisterUser(UserToRegister userToRegister)
        {
            wc.Headers.Add("Content-Type", "application/json");
            wc.Headers.Add("Data-Type", "application/json");
            string userJson = serializer.Serialize(userToRegister);
            wc.UploadString(wc.BaseAddress + "RegisterUser", "Post", userJson);
        }

        public void AddEvent(CalendarEvent eventToPost)
        {
            wc.Headers.Add("Content-Type", "application/json");
            wc.Headers.Add("Data-Type", "application/json");
            string eventJson = serializer.Serialize(eventToPost);
            wc.UploadString(wc.BaseAddress + "PostEvent", "Post", eventJson);
        }
    }
}

File: UberCalendarPythonClient/Controller/WebApiDataHandler.py
import requests
import json
import datetime
from Data.CalendarUser import CalendarUser
from Data.CalendarEvent import CalendarEvent

class WebApiDataHandler():
    def __init__(self):
        base_adress_port = 8080
        self.url = "http://localhost:" + str(base_adress_port) + "/UberCalendar/"
        iis_port = 8080
        test_port = 50925

    def CredentialsCheck(self,credentials):
        credentialsDictionary = { "Email": credentials.email, "Password": credentials.password}
        data = json.dumps(credentialsDictionary)
        request = requests.post(self.url + "GetUser",json = data)
        user_dictionary = dict(request.json())
        user  = CalendarUser(user_dictionary["UserID"],user_dictionary["Name"],user_dictionary["Surname"],user_dictionary["DateOfBirth"])
        return user

    def GetEvents(self,dateForEvents,userId):
        paramsForEvents = {"dateForEvents":str(dateForEvents), "loggedInUser":userId}
        testUrl = self.url + "GetEventsForUser?date=" + str(dateForEvents) + "&userId=" + str(userId)
        request = requests.get(testUrl)
        listOfEvents = list(request.json())
        events = list()
        for object in listOfEvents:
            events.append(CalendarEvent(object["Id"],object["Description"],object["UserId"],object["Date"],object["Time"],object["Type"]))
        return events

    def RegisterUser(self,userToRegister,credentialsToRegister):
        registerDictionary = {
            "Email": credentialsToRegister.email,
            "Password": credentialsToRegister.password,
            "Name": userToRegister.name,
            "Surname": userToRegister.surname,
            "DateOfBirth": userToRegister.date_of_birth}
        data = json.dumps(registerDictionary)
        request = requests.post(self.url + "RegisterUser", json = data)
        response = request.status_code
        print(response)

    def AddEvent(self,eventToAdd):
        eventDictionary = { "Description": eventToAdd.description, "userId": eventToAdd.user_id, "date": eventToAdd.date, "time": eventToAdd.time, "type": eventToAdd.type }
        data = json.dumps(eventDictionary)
        request = requests.post(self.url + "PostEvent",json=data)
        response = request.status_code

test = WebApiDataHandler()
dateTest = str(datetime.datetime(2019,1,30).date()) 
userIdTest = 29
test.GetEvents(dateTest,userIdTest)



